x/axelarnet/keeper/msg_server.go

```diff
@@ -80,11 +80,24 @@ func (s msgServer) CallContract(c context.Context, req *types.CallContractReques
 	payloadHash := crypto.Keccak256(req.Payload)
 
 	msgID, txID, nonce := s.nexus.GenerateMessageID(ctx)
-	msg := nexus.NewGeneralMessage(msgID, sender, recipient, payloadHash, nexus.Processing, txID, nonce, nil)
+	msg := nexus.NewGeneralMessage(msgID, sender, recipient, payloadHash, nexus.Approved, txID, nonce, nil)
 	if err := s.nexus.SetNewMessage(ctx, msg); err != nil {
 		return nil, sdkerrors.Wrap(err, "failed to add general message")
 	}
 
+	if req.Fee != nil {
+		events.Emit(ctx, &types.FeePaid{
+			MessageID: msgID,
+			Recipient: req.Fee.Recipient,
+			Fee:       req.Fee.Amount,
```

Comment:
What if amount is zero?

Reply:
No funds will be sent, but it shouldn't break anything

Reply:
No, I meant is axelaron supposed to filter it out?

Reply:
Yeah, it should be prevented during validation

Reply:
should validate the fee in `ValidateBasic` regardless

Reply:
ok added a check for this